src/decoder/stream.rs

```diff
@@ -770,22 +793,25 @@ impl StreamingDecoder {
                     self.current_chunk.type_ = type_str;
                     self.inflater.finish_compressed_chunks(image_data)?;
                     self.inflater.reset();
+                    self.ready_for_idat_chunks = false;
```

Comment:
Why is it set to `false` after `IDAT` too? There could be multiple consecutive `IDAT`s.

Reply:
Here we do not have **consecutive** `IDAT`s:

* At a high-level this piece of code handles transitioning from 1-or-more-IDAT-chunks to some-other-chunk.  For example - when that happens we flush the remaining decompressed data out of `self.inflater`).
* On old lines 767-768 (new lines 790-791) above the new `type_str` is not an `IDAT` anymore, because it is different from the previous chunk's `self.current_chunk.type_`.